### PR TITLE
Add support for predictable interface naming

### PR DESCRIPTION
**Pull request by w-yue**
What does this PR do?
We currently uses hard-coded "eth0" as nic interface in our code which prevents Mizar from running in system whose default interface is not named as eth0, which is the case in later Ubuntu version or other Linux flavors. 

This PR remove hard coded "eth0" from the logic and replaces with the default interface name obtained from the system. It is a fix for issue #519.

How was this tested?
It is tested in an AWS 2 nodes cluster environment with default interface name as "ensxx", the k8s cluster was launched successfully and the default interface name is detected and used in Mizar pods(instead of eth0), the testpods were created successfully and ping-able to each other.

Are there any user facing / API changes?
No.

### CODE REVIEW

**Comment by vinaykul** [comment]
It seems to be hung in kind-e2e-test. I need to debug my failure as I cannot get local repro. I'll quit this CI and restart later.

**Comment by vinaykul** [comment]
Found and fixed the issue that was causing CI failure in my PR. 
I've now restarted your CI.

File: mizar/common/common.py [hunk 1]
```diff
@@ -426,9 +426,24 @@ def conf_list_has_max_elements(conf, conf_list):
...
+        172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1
+        ...
+    """
+    default_itf="eth0"
+    ret, data = run_cmd("ip route")
+    data = [i for i in data.split('\n') if i]
+    for line in data:
+        if line.startswith('default'):
+            return line.split()[4]
+    # return  data[0].split()[2] # this should not be reached
```
Comment (vinaykul) [inline mizar/common/common.py:442]: Can we remove this line if it is not doing anything?

**Comment by vinaykul** [comment]
Please see if we need to fix the eth0 reference in scripts/03_start.sh

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ CFLAGS += -Wall
 CFLAGS += -Wextra
 CFLAGS += -Werror
 CFLAGS += -pedantic -Wpedantic
-CLFAGS += -Wno-cast-function-type -Wno-error=cast-function-type
+#CFLAGS += -Wno-cast-function-type -Wno-error=cast-function-type
```
Comment (vinaykul) [inline Makefile:47]: Please add a comment explaining why this is commented out and still left in the file.

File: mizar/daemon/app.py [hunk 3]
```diff
@@ -83,20 +86,22 @@ def init(benchmark=False):
...
         nsenter -t 1 -m -u -n -i ip link add {CONSTANTS.MIZAR_BRIDGE} type bridge && \
         nsenter -t 1 -m -u -n -i ip link set dev {CONSTANTS.MIZAR_BRIDGE} up && \
-        nsenter -t 1 -m -u -n -i ip link set eth0 master {CONSTANTS.MIZAR_BRIDGE} && \
+        nsenter -t 1 -m -u -n -i ip link set {default_itf} master {CONSTANTS.MIZAR_BRIDGE} && \
         nsenter -t 1 -m -u -n -i ip addr add {nodeip} dev {CONSTANTS.MIZAR_BRIDGE} && \
         nsenter -t 1 -m -u -n -i brctl show'''
 
-    rtlistcmd = 'nsenter -t 1 -m -u -n -i ip route list | grep "dev eth0"'
+    dev_default_itf = f'''dev {default_itf}'''
+    rtlistcmd = 'nsenter -t 1 -m -u -n -i ip route list | grep "' + f'''{dev_default_itf}''' +'"'
```
Comment (vinaykul) [inline mizar/daemon/app.py:94]: Nit: space after the last '+'

File: mizar/daemon/interface_service.py [hunk 4]
```diff
@@ -334,7 +334,7 @@ def ActivateHostInterface(self, interface):
 
 
 class LocalTransitRpc:
-    def __init__(self, ip, mac, itf='eth0', benchmark=False):
+    def __init__(self, ip, mac, itf, benchmark=False):
```
Comment (vinaykul) [inline mizar/daemon/interface_service.py:337]: Is there any issue if we kept the default parameter value?

File: pkg/util/netutil/netutil.go [hunk 5]
```diff
@@ -112,13 +112,13 @@ func ActivateInterface(
 	}
 
 	strBuilder.WriteString("\nDisable tso for pod")
-	cmdTxt, result, err := executil.Execute("ip", "netns", "exec", netNSFileName, "ethtool", "-K", "eth0", "tso", "off", "gso", "off", "ufo", "off")
+	cmdTxt, result, err := executil.Execute("ip", "netns", "exec", netNSFileName, "ethtool", "-K", ifName, "tso", "off", "gso", "off", "ufo", "off")
```
Comment (vinaykul) [inline pkg/util/netutil/netutil.go:115]: where's ifName set?

File: test/trn_controller/droplet.py [hunk 6]
```diff
@@ -9,18 +9,21 @@
...
+from mizar.common.common import *
 import os
 import docker
 import time
 import json
 
 
 class droplet:
-    def __init__(self, id, droplet_type='docker', phy_itf='eth0', benchmark=False):
+    def __init__(self, id, phy_itf, droplet_type='docker', benchmark=False):
```
Comment (vinaykul) [inline test/trn_controller/droplet.py:20]: any harm in keeping default parameter here?

File: mizar/daemon/app.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 from mizar.daemon.interface_service import InterfaceServer
 from mizar.daemon.droplet_service import DropletServer
 from mizar.common.constants import CONSTANTS
+from mizar.common.common import *
```
Comment (vinaykul) [inline mizar/daemon/app.py:11]: Do we need to import everything?

File: test/trn_controller/droplet.py [hunk 8]
```diff
@@ -9,18 +9,21 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 from test.trn_controller.common import cidr, logger, run_cmd
+from mizar.common.common import *
```
Comment (vinaykul) [inline test/trn_controller/droplet.py:12]: same here.

**Comment by w-yue** [comment]
Close this PR. Will resubmit after CI issue investigation.

**Comment by w-yue** [comment]
https://github.com/CentaurusInfra/mizar/pull/555 is the new PR for the same issue.

**Pull request closed without merging**